### DISCUSSION TITLE
Download Audio Tracks Only

### DISCUSSION THREAD

**Discussion by lesclaz** [crunchy-labs/crunchy-cli#366] [Ideas]
I believe it would be extremely useful to be able to download only the audio tracks, since anime on Crunchyroll are usually initially published in Japanese and then dubbed into other languages over time. It seems unnecessary, as well as inefficient, to have to download a complete file that includes the video track when it is really the audio that interests me. Moreover, repeatedly downloading the same video tracks results in excessive data consumption.
Do you think you could consider creating such a feature?

**Comment by bytedream** [comment]
I will think about it. I'm limited with my time for this project rn, so it doesn't have priority.
But with a bit of manual work you can use the search command:
```shell
$ crunchy-cli -q search --audio en-US -o "{{stream.dash_url}}" https://www.crunchyroll.com/watch/GRDKJZ81Y/alone-and-lonesome
```
This will return the dash stream url, which contains video and audio, for the given `--audio` language

**Reply by lesclaz** [reply]
I had already read something about that in an issue, I tried it and it didn't work for me. When using wget or opening the provided URL directly from the browser, I get an authorization error. Furthermore, this still has the same data consumption issue as downloading the full video, am I wrong?

I want to thank you for the excellent work you do.

**Reply by bytedream** [reply]
I've added the `{{account.token}}` output option to `search`. With this you are able to get the token which is needed to request the url. Upon request you then need to set the header `Authorization: Bearer <token>`.
I hadn't implemented it prior since the need for authorization to request a stream is only in place since the new endpoint update Crunchyroll did
